src/endpoint/s3/s3_utils.js

```diff
@@ -131,9 +131,19 @@ function get_request_xattr(req) {
 }
 
 function set_response_xattr(res, xattr) {
-    _.each(xattr, (val, key) => {
-        res.setHeader(X_AMZ_META + key, val);
-    });
+    let size_for_md_left = config.S3_MD_SIZE_LIMIT;
+    const sorted_keys = Object.keys(xattr).sort();
+    let returned_keys = 0;
+    for (const key of sorted_keys) {
+        const property_size = Buffer.byteLength(key, 'utf8') + Buffer.byteLength(xattr[key], 'utf8');
```

Comment:
I think you should include the actual http header size, which is: `X_AMZ_META + key + ': ' + value + '\n'`
which means:
```suggestion
        const md_header_size = 
            X_AMZ_META.length +
            3 + // for ': ' and '\n'
            Buffer.byteLength(key, 'utf8') +
            Buffer.byteLength(xattr[key], 'utf8');
```